### PR TITLE
Fix env configuration and auth/user controller updates

### PR DESCRIPTION
**Pull request by pbn4**
Addresses https://github.com/bloom-housing/bloom/issues/758:

- user creation moved to UserController,
- configuration refactored - ConfigService introduced,
- user dtos refactored,
- existing tests were fixed,
- introduces global ClassSerializerInterceptor and maps classes into response dtos in each controller.

While I was trying to move user creation over from auth module to user module I have hit a weird problem with process not starting and complaining about SendGrid Api Key not being defined (triggered when User module tries to import SharedModule). Turns out we were loading .env in ormconfig.ts and that introduced file import ordering dependency. 

I introduced ConfigService from `@nestjs/config` to make sure modules use env variables when they are loaded from `.env`. This is not applied for ormconfig.ts and ormconfig.test.ts because I don't currently see how to solve having ormconfig use ConfigService outside AppModule.

### CODE REVIEW

File: backend/core/src/auth/login.dto.ts [hunk 1]
```diff
@@ -2,17 +2,16 @@ import { IsEmail, IsString } from "class-validator"
 import { Expose } from "class-transformer"
 
 export class LoginDto {
+  @Expose()
   @IsEmail()
   email: string
 
+  @Expose()
```
Comment (software-project) [inline backend/core/src/auth/login.dto.ts:9]: Password should be exposed?

File: backend/core/src/user/user.dto.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  @ApiHideProperty()
+  createdAt
+
+  @Exclude()
+  @ApiHideProperty()
+  updatedAt
+
+  @Expose()
+  @IsString()
+  password: string
```
Comment (software-project) [inline backend/core/src/user/user.dto.ts:36]: Same here with expose.

**Review by software-project — APPROVED** [review]
Couple small questions, but otherwise it looks good.

File: backend/core/src/app.module.ts [hunk 3]
```diff
@@ -34,6 +45,15 @@ export function applicationSetup(app: INestApplication) {
 
 @Module({
   imports: [
+    ConfigModule.forRoot({
+      validationSchema: Joi.object({
+        PORT: Joi.number().default(3100).required(),
```
[the remark this answers is not in the document]
Comment (pbn4) [reply backend/core/src/app.module.ts:50]: Config module is loading .env file. This is a validation definition for those env variables. Default means that when there is no env var a 3100 will be used.

File: backend/core/src/auth/auth.controller.ts [hunk 4]
```diff
@@ -33,24 +21,14 @@ export class AuthController {
...
-    return new LoginResponseDto({ accessToken })
-  }
-
-  @Post("register")
-  @ApiOperation({ summary: "Register", operationId: "register" })
-  async register(@Body() params: CreateUserDto) {
-    const user = await this.userService.createUser(params)
-    const accessToken = this.authService.generateAccessToken(user)
-    // noinspection ES6MissingAwait
-    this.emailService.welcome(user)
```
[the remark this answers is not in the document]
Comment (pbn4) [reply backend/core/src/auth/auth.controller.ts:45]: It has been moved to UserController

File: backend/core/src/auth/login.dto.ts (hunk 1 above)
Comment (pbn4) [reply backend/core/src/auth/login.dto.ts:9]: Yup it's a LoginDTO

File: backend/core/src/user/user.dto.ts (hunk 2 above)
Comment (pbn4) [reply backend/core/src/user/user.dto.ts:36]: It's UserCreate DTO it must have a password prop

**Pull request merged**